app/presenters/comment_presenter.rb

```diff
@@ -0,0 +1,46 @@
+class CommentPresenter < ApplicationPresenter
+  presents :comment
+  delegate :text, to: :comment
+
+  def submitters_name
+    comment.submitter.name
+  end
+
+  def submitters_link
+    if h.user_signed_in?
+      h.comment_reply_comment_path(comment)
+    else
+      h.user_path(comment.submitter.name)
+    end
+  end
+
+  def link_prefix
+    if h.user_signed_in?
+      "Reply to "
+    else
+      " - contributed by "
+    end
+  end
+
+  def text_trailer
+    h.link_to (link_prefix + submitters_name), submitters_link
+  end
```

Comment:
The amount of methods / conditionals could be simplified with the below:

```
def attribution_link
  text, url = if h.user_signed_in?
    ["Reply to #{submitter_name}", h.comment_reply_comment_path(comment)]
  else
    [" - contributed by #{submitter_name}", h.user_path(submitter_name)]
  end

  h.link_to text, url
end
```

I also renamed it to `attribution_link` because that felt like it better described what this returns. Thoughts?

Reply:
I like that - though I used `string` and `link` instead of `text` and `url`. `text` and `url` are the types of posts one can submit, so its seems confusing to use the same terms here.

Reply:
cool. related to the comment about using the same terms, you may find the DDD concept of [bounded contexts](https://martinfowler.com/bliki/BoundedContext.html) interesting.